src/main/java/cart/controller/ProductController.java

```diff
@@ -0,0 +1,46 @@
+package cart.controller;
+
+import cart.domain.Product;
+import cart.dto.ProductCreateDto;
+import cart.dto.ProductUpdateDto;
+import cart.service.ProductService;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ProductController {
+
+  private final ProductService productService;
+
+  public ProductController(ProductService productService) {
+    this.productService = productService;
+  }
+
+  @GetMapping("/products")
+  public ResponseEntity<List<Product>> getAll() {
+    List<Product> products = productService.getAll();
+    return ResponseEntity.ok(products);
+  }
+
+  @PutMapping(value = "/products")
+  public void updateProduct(@RequestBody ProductUpdateDto updateDto) {
+    productService.updateProduct(updateDto);
+  }
```

Comment:
변경할 대상에 대해 pathVariable 로 명시하는 편인거 같아요.
https://gmlwjd9405.github.io/2018/09/21/rest-and-restful.html

Reply:
@Hyeon9mak 추가 질문드립니다!
해당 기능 구현시 pathVariable도 고민했었는데, 
updateProduct메서드에서 pathVariable를 파라미터로 받아서 메소드 내에서 파라미터를 따로 쓰고있지않아서 
ProductUpdateDto만 넘겨받도록처리했습니다.

그래도 url에서 변경할 대상을 명시하는것에 좋을가요~?

Reply:
```suggestion
  @PutMapping(value = "/products/{id}")
  public void updateProduct(@PathVariable Long id,
                            @RequestBody ProductUpdateDto updateDto) {
    productService.findById(id);
    productService.updateProduct(updateDto);
  }
```

- pathVariable로 url에 명시하도록 변경하였습니다.
- 해당 상품이있는지 확인하는 로직을 추가하여 파라미터로 받은 id를 사용하도록했습니다.
- `productService.findById(id);` 부분은 productService.updateProduct(updateDto);의 내부로 넣는게나을까요? 지금처럼 컨트롤러에서 확인해도 괜찮을지 궁금합니다!